components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/user/impl/UserInfoJSONResponseBuilder.java

```diff
@@ -60,10 +64,18 @@ public String getResponseString(OAuth2TokenValidationResponseDTO tokenResponse)
             throws UserInfoEndpointException {
         Resource resource = null;
         try {
+            PrivilegedCarbonContext.startTenantFlow();
+            PrivilegedCarbonContext carbonContext = PrivilegedCarbonContext.getThreadLocalCarbonContext();
+            int tenantId = IdentityTenantUtil.getTenantIdOfUser(tokenResponse.getAuthorizedUser());
```

Comment:
Can we assume we get tenant qualified username in all cases like federated users, or saas apps?